helm-charts/azure-api-management-gateway/templates/service.yaml

```
@@ -10,6 +10,9 @@ metadata:
   {{- end }}
 spec:
   type: {{ .Values.service.type }}
+{{- if (.Values.service.loadBalancer).ip }}
```

Is this a bug or not?

I think this should be:
```suggestion
{{- if (.Values.service.loadBalancer.ip) }}
```

But I might be wrong here and not be aware of a Helm template feature.

---

Should be as in the commit. The parentheses make sure the `service.loadBalancer` doesn't throw a nil pointer if it's not defined.

We can omit the parentheses and add a default value for `loadBalancer` in values.yaml if that's better:
```
service:
  type: ClusterIP
  loadBalancer: {}
```

---

You learn something new every day - Thank you!